cdap-ui/app/cdap/components/Replicator/Detail/ThroughputGraph/throughput.ts

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as d3 from 'd3';
+import numeral from 'numeral';
+import { objectQuery } from 'services/helpers';
+
+export const COLOR_MAP = {
+  inserts: '#185ABC',
+  updates: '#669DF6',
+  deletes: '#AECBFA',
+  horizontalLine: '#DDDDDD',
+  verticalLine: '#979797',
+  legend: '#999999',
+  tick: '#333333',
+};
+
+export interface IThroughputData {
+  time: number;
+  inserts: number;
+  updates: number;
+  deletes: number;
+  errors: number;
+}
+
+const timeFormat = d3.timeFormat('%m/%d %I:%M %p');
```

Comment:
We should move our formatting functions to a shared implementation.

Reply:
FWIW, there's a section in the UI component proposal for formatters: go/cdf-ui-component-design I'll flesh that out some more.